File: c/moveZeroes.c
#include <stdio.h>

void swap (int *nums , const int left, const int right) {
    const int temp = *(nums + left);
    *(nums + left) = *(nums + right);
    *(nums + right) = temp;
}

void moveZeroes(int* nums, int numsSize) {
    int left = 0, right = 0;
    while (right < numsSize) {
        if (nums[right] != 0) {
            swap(nums, left, right);
            left++;
        }
        right++;
    }
}

void test() {
    int nums[] = {0, 1, 0, 3, 12};
    int numsSize = 5;
    moveZeroes(nums, numsSize);
    printf("[");
    for (int i = 0; i < numsSize; i++) {
        printf("%d", *(nums + i));
        if (i == numsSize - 1) {
            printf("]");
        } else {
            printf(", ");
        }
    }
}

File: src/leetcode/movezeroes.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#
from typing import List

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left, right = 0, 0
        for right in range(len(nums)):
            if nums[right] != 0:
                nums[right], nums[left] = nums[left], nums[right]
                left += 1
        
# @lc code=end

def main():
    sol = Solution()
    nums = [0, 1]
    sol.moveZeroes(nums)
    print(nums)
    
if __name__ == "__main__":
    main()
